coverage.py

```diff
@@ -358,13 +364,15 @@ def set_metadata_and_circulation_data(self, identifier, metadata, circulationdat
         CoverageFailure (if not).
         """
 
-        result = self._set_metadata(identifier, metadata, metadata_replacement_policy)
-        if isinstance(result, CoverageFailure):
-            return result
-        
-        result = self._set_circulationdata(identifier, circulationdata, circulationdata_replacement_policy)
-        if isinstance(result, CoverageFailure):
-            return result
+        if metadata:
+            result = self._set_metadata(identifier, metadata, metadata_replacement_policy)
+            if isinstance(result, CoverageFailure):
+                return result
+
+        if circulationdata:
```

Comment:
Calling set_circulationdata when no CirculationData was provided was causing an error, which resulted in an incorrect CoverageFailure.